yb-voyager/src/query/queryissue/parser_issue_detector_test.go

```diff
@@ -2341,3 +2342,157 @@ func TestMissingForeignKeyIndexDetection(t *testing.T) {
 		assert.True(t, cmp.Equal(expectedIssue, issues[0]), "Expected issue not found: %v\nFound: %v", expectedIssue, issues[0])
 	})
 }
+
+// ====== Primary Key Recommendation Testing - Checks for missing PK when UNIQUE and NOT NULL columns are available ======
+
+func runPKRec(t *testing.T, name string, ddls []string, expected []QueryIssue) {
+	// Helper function to build parser issue detector from DDL statements
+	pkrecBuild := func(ddls ...string) *ParserIssueDetector {
+		d := NewParserIssueDetector()
+		for _, s := range ddls {
+			if s == "" {
+				continue
+			}
+			_ = d.ParseAndProcessDDL(s)
+		}
+		d.FinalizeColumnMetadata()
+		return d
+	}
+
+	// Helper function to create a key for sorting QueryIssues
+	pkrecKey := func(q QueryIssue) string {
+		opt, _ := q.Details["PrimaryKeyColumnOptions"].([][]string)
+		// Sort the column options to ensure consistent ordering
+		sortedOpt := make([][]string, len(opt))
+		copy(sortedOpt, opt)
+		slices.SortFunc(sortedOpt, func(a, b []string) int {
+			if len(a) != len(b) {
+				return len(a) - len(b)
+			}
+			return strings.Compare(strings.Join(a, ","), strings.Join(b, ","))
+		})
+		// Create a string representation for the key
+		keyParts := make([]string, len(sortedOpt))
+		for i, cols := range sortedOpt {
+			keyParts[i] = strings.Join(cols, ",")
+		}
+		joined := strings.Join(keyParts, "|")
+		return q.ObjectName + "|" + joined
+	}
+
+	// Helper function to sort QueryIssues for consistent comparison
```

Comment:
check out `assert.ElementsMatch()` . I think that'd avoid having to do all of this.